ci/lava/helpers.py

```diff
@@ -0,0 +1,340 @@
+    """Get the kernel version running on the DUT using mbl-cli."""
+    exit_code, output, error = execute_helper.send_mbl_cli_command(
+        ["shell", 'su -l -c "uname -a"'], dut_addr
+    )
+    return exit_code, output
+
+
+def get_file_contents_md5(path, dut_addr, execute_helper):
+    """Run md5sum on a file on the DUT."""
+    exit_code, output, error = execute_helper.send_mbl_cli_command(
+        ["shell", 'su -l -c "md5sum {}"'.format(path)],
+        dut_addr,
+        raise_on_error=True,
+    )
+    return output.split("\n")[1].split()[0]
+
+
+def get_file_sha256sum(path, dut_addr, execute_helper):
+    """Run sha256sum on a file on the DUT."""
+    exit_code, output, error = execute_helper.send_mbl_cli_command(
+        ["shell", 'su -l -c "sha256sum {}"'.format(path)],
+        dut_addr,
+        raise_on_error=True,
+    )
+    return output.split("\n")[1].split()[0]
+
+
+def get_mounted_bank_label(mount_point, dut_addr, execute_helper):
+    """Get the mounted partition label at a particular mount."""
+    exit_code, output, error = execute_helper.send_mbl_cli_command(
+        ["shell", '/usr/bin/lsblk --noheadings --output "MOUNTPOINT,LABEL"'],
+        dut_addr,
+        raise_on_error=True,
+    )
+    for line in output.split("\n"):
+        match = re.search(r"^{} .*".format(mount_point), line)
+        if match:
+            return match.group(0).split()[1]
+
+    raise AssertionError(
+        "No match for the mount point found! Output was: {}".format(output)
+    )
+
+
+def get_file_mtime(path, dut_addr, execute_helper):
+    """Get the last modified time of a file on the DUT."""
+    exit_code, output, error = execute_helper.send_mbl_cli_command(
+        ["shell", "stat {}".format(path)], dut_addr, raise_on_error=True
+    )
+    for line in output.split("\n"):
+        if "Modify:" in line:
+            mtime = line
+            break
+    if not mtime:
+        raise AssertionError(
+            "Could not find Modified Time in the output from `stat`. "
+            "The output was {}".format(output)
+        )
+    return mtime
+
+
+def strings_grep(dut_addr, execute_helper, file_path, pattern):
+    """Run strings command on a file and grep for a pattern using mbl-cli."""
+    command = "strings {} | grep {}".format(file_path, pattern)
+    exit_code, output, error = execute_helper.send_mbl_cli_command(
+        ["shell", 'sh -l -c "{}"'.format(command)],
+        dut_addr,
+        raise_on_error=True,
+    )
+    return output.split("\n")[1]
+
+
+class ExecuteHelper:
+    """Class to provide a wrapper for executing commands as a subprocess."""
+
+    debug = False
+
+    def __init__(self, request):
+        """Initialise the class."""
+        ExecuteHelper.debug = request.config.getoption("--debug-output")
+
+    @staticmethod
+    def _print(data):
+        if ExecuteHelper.debug:
+            print(data)
+
+    @staticmethod
+    def execute_command(command):
```

Comment:
In warrior-dev conftest.py this method has been changed to support a timeout value. Because this class has moved into a new file this is not obvious.
Please apply the changes manually from warrior-dev.

Reply:
Will do.